x-pack/legacy/plugins/epm/server/lib/template/template.ts

```diff
@@ -23,7 +25,34 @@ export interface Template {
   aliases: object;
 }
 
-function getBaseTemplate(): Template {
+/**
+ * Generate mapping takes the given fields array and creates the Elasticsearch
+ * mapping properties out of it.
+ *
+ * @param fields
+ */
+export function generateMappings(fields: Field[]): object {
+  const props = {};
+  fields.forEach(field => {
+    // Are there more fields inside this field? Build them recursively
+    if (field.fields && field.fields.length > 0) {
+      props[field.name] = generateMappings(field.fields);
+      return;
+    }
+
+    // If not type is defined, take keyword
+    let t = 'keyword';
+    if (field.type) {
+      t = field.type;
+    }
+    props[field.name] = {
+      type: t,
```

Comment:
Or, since we don't use `t`

```suggestion
      type: field.type || 'keyword',
```